tasks/eth/ink-001-permissionless-proofs/NestedSignFromJson.s.sol

```diff
@@ -0,0 +1,191 @@
+        require(uint64(Duration.unwrap(currentImpl.clockExtension())) == uint64(Duration.unwrap(faultDisputeGame.clockExtension())), "80");
+
+        if (_targetGameType.raw() == GameTypes.PERMISSIONED_CANNON.raw()) {
+            PermissionedDisputeGame currentPDG = PermissionedDisputeGame(address(currentImpl));
+            PermissionedDisputeGame permissionedDisputeGame = PermissionedDisputeGame(address(faultDisputeGame));
+            require(address(currentPDG.proposer()) == address(permissionedDisputeGame.proposer()), "90");
+            require(address(currentPDG.challenger()) == address(permissionedDisputeGame.challenger()), "100");
+        }
+    }
+
+    function _precheckAnchorStateCopy(GameType _fromType, GameType _toType) internal view {
+        console.log("pre-check anchor state copy", _toType.raw());
+
+        FaultDisputeGame fromImpl = FaultDisputeGame(address(dgfProxy.gameImpls(GameType(_fromType))));
+        // Must have existing game type implementation for the source
+        require(address(fromImpl) != address(0), "200");
+        address fromRegistry = address(fromImpl.anchorStateRegistry());
+        require(fromRegistry != address(0), "210");
+
+        FaultDisputeGame toImpl = FaultDisputeGame(address(dgfProxy.gameImpls(GameType(_toType))));
+        if (address(toImpl) != address(0)) {
+            // If there is an existing implementation, it must use the same anchor state registry.
+            address toRegistry = address(toImpl.anchorStateRegistry());
+            require(toRegistry == fromRegistry, "210");
+        }
+    }
+
+    function getAllowedStorageAccess() internal view override returns (address[] memory allowed) {
+        allowed = new address[](5 + extraStorageAccessAddresses.length);
+        allowed[0] = address(dgfProxy);
+        allowed[1] = address(ownerSafe);
+        allowed[2] = address(councilSafe);
+        allowed[3] = address(foundationSafe);
+        address livenessGuard = address(uint160(uint256(vm.load(address(councilSafe), livenessGuardSlot))));
+        allowed[4] = livenessGuard;
+
+        for (uint256 i = 0; i < extraStorageAccessAddresses.length; i++) {
+            allowed[5 + i] = extraStorageAccessAddresses[i];
+        }
+        return allowed;
+    }
+
+    /// @notice Checks the correctness of the deployment
+    function _postCheck(Vm.AccountAccess[] memory accesses, Simulation.Payload memory) internal view override {
+        console.log("Running post-deploy assertions");
+
+        checkStateDiff(accesses);
+        _postcheckAnchorStateCopy(GameType.wrap(0), bytes32(0x5220f9c5ebf08e84847d542576a67a3077b6fa496235d93c557d5bd5286b431a), 523052);
+        // Removing this because the current anchor state on ink mainnet is 0
+        // This is because the chain is currently less than 7 days old and hasn't had a chance to finalize any games.
+        // _postcheckHasAnchorState(GameType.wrap(1));
```

Comment:
Let's uncomment before the signing windows starts

Reply:
done